pyrsia_node/src/block_chain/block_chain.rs

```diff
@@ -0,0 +1,174 @@
+/*
+   Copyright 2021 JFrog Ltd
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use log::{info, warn};
+use serde_json::error::Error;
+use sha2::{Digest, Sha256};
+
+use super::block::Block;
+
+pub struct BlockChain {
+    blocks: Vec<Block>,
+}
+
+const DIFFICULTY_PREFIX: &str = "00";
+
+impl BlockChain {
+    pub fn new() -> Self {
+        BlockChain { blocks: vec![] }
+    }
+    pub fn genesis(&mut self) -> &mut Self {
+        let genesis_block = Block {
+            id: 0,
+            timestamp: now(),
+            previous_hash: String::from("genesis"),
+            data: String::from("genesis!"),
+            nonce: 2836,
+            hash: "0000f816a87f806bb0073dcf026a64fb40c946b5abee2573702828694d5b4c43".to_string(),
+        };
+        self.blocks.push(genesis_block);
+        self
+    }
+    pub(crate) fn dump(&self) -> Result<String, Error> {
```

Comment:
Is the crate context needed?

```suggestion
    pub fn dump(&self) -> Result<String, Error> {
```

Reply:
@prince-chrismc Why are you concerned about the crate scope? This appears to be a method to use for dumping out the contents of the blockchain. At this point, it is perfectly reasonable to not want to expose it as part of the Blockchain's public API.